### PR TITLE
added support for having screenshots of website

### PR DESCRIPTION
**Pull request by Abhimanyu121**

### CODE REVIEW

File: src/components/Deployment/service.ts [hunk 1]
```diff
@@ -62,7 +63,26 @@ const DeploymentService: IDeploymentService = {
         }
 
         return DeploymentModel.findOneAndUpdate(condition, update);
+    },
+    
+    async updateScreenshotUrl(deploymentId: string, screenshotUrl: IScreenshot): Promise<IDeployment> {
```
Comment (akinovak) [inline src/components/Deployment/service.ts:68]: This should be called just screenshot instead of "screenshotUrl"

Comment (Abhimanyu121) [reply src/components/Deployment/service.ts]: Done

File: src/components/Deployment/service-interface.ts [hunk 2]
```diff
@@ -9,4 +9,6 @@ export interface IDeploymentService {
     find(query: Partial<IDeployment>): Promise<Array<IDeployment>>;
     updateFinishedDeployment(deploymentId: string, sitePreview: string, deploymentStatus: string, buildTime: number, logs: Array<string>): Promise<IDeployment>;
     updatePayment(deploymentId: string, paymentId: string): Promise<IDeployment>;
+    updateScreenshot(deploymentId: string, screenshotUrl: IScreenshot): Promise<IDeployment>;
```
Comment (akinovak) [inline src/components/Deployment/service-interface.ts:12]: Here also :)

File: src/config/env/index.ts [hunk 3]
```diff
@@ -29,6 +29,7 @@ interface IConfig {
   };
   arweave: {
     LOGSTOCAPTURE: Array<any>;
+    key: string;
```
Comment (rekpero) [inline src/config/env/index.ts:32]: Rename it to `PRIVATE_KEY`, full caps

File: src/components/Deployment/service.ts [hunk 4]
```diff
@@ -62,7 +63,26 @@ const DeploymentService: IDeploymentService = {
...
+        const update = {
+            screenshotData,
+        }
+
+        return DeploymentModel.findOneAndUpdate(condition, update);
+    },
+    async uploadScreenshotToArweave(url: string): Promise<IScreenshot>{
+        const nftServices: nftlib.Services = new nftlib.Services(config.arweave.key)
+        const nft: nftlib.Nft = new nftlib.Nft(undefined, nftServices)
+        const screenshoObj: IScreenshot = await nft.uploadScreenshotToArweave(url)
```
Comment (rekpero) [inline src/components/Deployment/service.ts:82]: don't need to append with Obj, just name it `screenshot`

File: package.json [hunk 5]
```diff
@@ -36,6 +36,8 @@
     "node": ">=10.0.0 <15.0.0"
   },
   "dependencies": {
+    "@argoapp/nft-js": "git+https://github.com/argoapp-live/argo-nft-js.git",
+    "@argoapp/payment-js": "git+https://github.com/argoapp-live/argo-nft-js.git",
```
Comment (rekpero) [inline package.json:40]: why is this here

**Review by rekpero — CHANGES_REQUESTED** [review]

File: src/components/Deployment/model.ts [hunk 6]
```diff
@@ -34,6 +40,11 @@ const DeploymentSchema: Schema = new Schema(
       type: Schema.Types.ObjectId,
       ref: "ProjectModel",
     },
+    screenshotData: {
```
Comment (rekpero) [inline src/components/Deployment/model.ts:43]: No need to append Data, it's self understandable

File: src/components/Deployment/model.ts [hunk 7]
```diff
@@ -15,6 +15,12 @@ export interface IDeployment extends Document {
   project: IProject["_id"];
   createdAt: any;
   updatedAt: any;
+  screenshotData: any;
```
Comment (rekpero) [inline src/components/Deployment/model.ts:18]: Here too, also create an interface rather than using `any`

**Review by rekpero — CHANGES_REQUESTED** [review]

**Review by rekpero — APPROVED** [review]

**Pull request merged**